main.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/scylladb/go-set/strset"
+	"github.com/spf13/pflag"
+	"go.uber.org/zap"
+)
+
+var (
+	validFormats = strset.New("png", "pdf")
+	skipDirs     = strset.New(".git")
+
+	// register the log-level flag natively to the stdlib flag package
+	logLevel = zap.LevelFlag("log-level", zap.InfoLevel, "What log level to use")
+)
+
+func main() {
+	// Setup logger
+	log := zap.NewExample().WithOptions(zap.IncreaseLevel(logLevel))
+	defer func() { _ = log.Sync() }()
+	zap.ReplaceGlobals(log)
+	if err := run(); err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func run() error {
+	// Setup default config
+	cfg := &Config{
+		RootDir: "/files",
+		Formats: []string{"png", "pdf"},
+		SubDirs: []string{"."},
+		Files:   map[string]string{},
+	}
+
+	// Unconditionally overwrite root path if GH Action workspace is set
+	if ghWorkspace := os.Getenv("GITHUB_WORKSPACE"); len(ghWorkspace) != 0 {
+		cfg.RootDir = ghWorkspace
+	}
+	// Register the --log-level flag from the stdlib flag
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	// Add and parse flags
+	addFlags(cfg)
+	pflag.Parse()
+
+	// Validate the config
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+
+	return render(cfg)
+}
+
+func addFlags(cfg *Config) {
+	pflag.StringVarP(&cfg.RootDir, "root-dir", "r", cfg.RootDir, "Where the root directory for the files that should be converted are. If the env var GITHUB_WORKSPACE is set, this flag will automatically be set to that value.")
+	pflag.StringSliceVar(&cfg.Formats, "formats", cfg.Formats, "Comma-separated list of formats to render the *.drawio files as, for use with --subdirs")
+	pflag.StringSliceVarP(&cfg.SubDirs, "subdirs", "d", cfg.SubDirs, "Comma-separated list of sub-directories of --root-dir to recursively search for *.drawio files to convert")
+	pflag.StringToStringVarP(&cfg.Files, "files", "f", cfg.Files, fmt.Sprintf("Comma-separated list of files to convert, of form 'dest-file=src-file'. The extension for src-file can be anything, while the dest-file extension must be either %s", validFormats))
+}
+
+type Config struct {
```

Comment:
extract into a separate package, so that it can be re-used by e.g. a plantuml generator?